plugins/bulk-renamer/RenamerDialog.vala

```diff
@@ -0,0 +1,171 @@
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  Authors:
+ *  Vartan Belavejian <https://github.com/VartanBelavejian>
+ *  Jeremy Wootten <jeremywootten@gmail.com>
+ *
+*/
+
+public class Files.RenamerDialog : Gtk.Dialog {
+    private Files.Renamer renamer;
+    private Gtk.Widget rename_button;
+    public RenamerDialog (string _basename, GLib.File[] _files) {
+        renamer.add_files (_files);
+        renamer.set_base_name (_basename);
+    }
+
+    construct {
+        deletable = true;
+        set_title (_("Bulk Renamer"));
+        var cancel_button = add_button (_("Cancel"), Gtk.ResponseType.CANCEL);
+
+        rename_button = add_button (_("Rename"), Gtk.ResponseType.APPLY);
+        rename_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+
+        renamer = new Renamer ();
+        renamer.margin = 12;
+        renamer.set_sort_order (RenameSortBy.NAME, false);
+        renamer.bind_property ("can-rename",
+                                rename_button, "sensitive",
+                                GLib.BindingFlags.DEFAULT | GLib.BindingFlags.SYNC_CREATE);
+
+        get_content_area ().add (renamer);
+
+        response.connect ((response_id) => {
+            switch (response_id) {
+                case Gtk.ResponseType.APPLY:
+                    if (renamer.can_rename) {
+                        try {
+                            renamer.rename_files ();
+                        } catch (Error e) {
+                            var dlg = new Granite.MessageDialog (
+                                "Error renaming files",
+                                e.message,
+                                new ThemedIcon ("dialog-error")
+                            );
+                            dlg.run ();
+                            dlg.destroy ();
+                        }
+                    }
+
+                    break;
+
+                default:
+                    close ();
+                    break;
+            }
+        });
+
+        key_press_event.connect ((event) => {
+            var mods = event.state & Gtk.accelerator_get_default_mod_mask ();
+            bool control_pressed = ((mods & Gdk.ModifierType.CONTROL_MASK) != 0);
+            bool other_mod_pressed = (((mods & ~Gdk.ModifierType.SHIFT_MASK) & ~Gdk.ModifierType.CONTROL_MASK) != 0);
+            bool only_control_pressed = control_pressed && !other_mod_pressed; /* Shift can be pressed */
+
+            uint keyval = map_key (event);
+
+            switch (keyval) {
+                case Gdk.Key.Escape:
+                    if (mods == 0) {
+                        response (Gtk.ResponseType.REJECT);
+                    }
+                    break;
+                case Gdk.Key.Return:
+                    if (mods == 0 && renamer.can_rename) {
+                        response (Gtk.ResponseType.APPLY);
```

Comment:
This definitely isn't the right way to do this. the "Rename" button should have `has_default = true` and entry widgets in the dialog should have `activates_default = true`